src/gui/src/mainWindow.cpp

```diff
@@ -757,11 +775,24 @@ void MainWindow::deleteRuler(const std::string& name)
   }
 }
 
+int MainWindow::requestHighlightGroup()
+{
+  HighlightGroupDialog dlg;
+  dlg.exec();
+  return dlg.getSelectedHighlightGroup();
+}
+
 void MainWindow::updateHighlightedSet(const QList<const Selected*>& items,
                                       int highlight_group)
 {
-  if (highlight_group >= 7)
+  if (highlight_group < 0) {
```

Comment:
How do you trigger this condition from the UI?  I've always found the current setup clunky with fixed numbered highlights.

Reply:
Never mind - I see this is choosing a highlight not adding a new set.